### DISCUSSION TITLE
Lifetime flight hours

### DISCUSSION THREAD

**Discussion by yifeim** [EdgeTX/edgetx#3276] [Ideas]
How do I log the total time / number of connections I have made to a model / a model tag / or by the radio at all? I'd like to know my play time similar to what steam does. This counter has to be persistent through power cycles. Thanks.

**Comment by pfeerick** [comment]
Use a persistent timer, and set the reset condition to manual reset. Use the same triggers that you would use to detect a normal flight. 

[For B&W Radios](https://edgetx.gitbook.io/edgetx-user-manual/b-and-w-radios/model-select/setup#timer-1-mode-switch)
[For Color screen Radios](https://edgetx.gitbook.io/edgetx-user-manual/edgetx-user-manual/user-manual-for-color-screen-radios/model-settings/model-setup/timer-1-2-3)

If you use a color screen radio, you can also use Offers's Flights flight counter widget, which is included in the EdgeTX SD card pack...
https://github.com/offer-shmuely/edgetx-x10-widgets/wiki/Flights

**Reply by yifeim** [reply]
This looks super cool! Will try and thanks.

**Comment by rlisch** [comment]
I am still looking for this total flight time counter. The provided persistent counter only makes sense if it could be adjusted (through GV) to the real/actual total flight time of the model which I keep in my case, semi-automatic, using Flight Logs in an Excel file.
Use it for maintenance of my turbines, gas engines and helis, very useful actually.

**Comment by offer-shmuely** [comment]
please note that the newer version, keep the (automatically updated) flight count both on the GV9, as well as on the sd-card, on a csv file,
that can later view using the FlightHistory viewer

https://github.com/EdgeTX/edgetx-sdcard/blob/master/sdcard/c480x272/SCRIPTS/TOOLS/FlightsHistory.lua

https://github.com/offer-shmuely/edgetx-x10-scripts/wiki/FlightsHistory
![image](https://github.com/EdgeTX/edgetx/assets/7026911/6b0d5ce6-fcef-4e73-957a-115a3bc1b2a3)

**Comment by rlisch** [comment]
Thanks Offer!Love the flight counter, installed it for all my models.It would be so nice to have also a Total flight hours counter where flight time can actually be adjusted to real flight time of each modelActually this is not possible with the persistent timer.I only could find one script for OpenTx on the Net and it works, more or less with EdgeTx but only starts automatically if an altitude sensor is installed and the Widget layout is poor and full screen only.Flying Helis and larger gas engines I use Total time for maintenance purpose as well for my Large Scale Gliders.Thanks againRichardOn Jun 17, 2023, at 5:16 PM, Offer Shmuely ***@***.***> wrote:﻿
please note that the newer version, keep the (automatically updated) flight count both on the GV9, as well as on the sd-card, on a csv file,
that can later view using the FlightHistory viewer
https://github.com/EdgeTX/edgetx-sdcard/blob/master/sdcard/c480x272/SCRIPTS/TOOLS/FlightsHistory.lua
https://github.com/offer-shmuely/edgetx-x10-scripts/wiki/FlightsHistory

**Reply by offer-shmuely** [reply]
I will think about it...

**Comment by offer-shmuely** [comment]
I maybe can show the total time on this page:


![image](https://github.com/EdgeTX/edgetx/assets/7026911/6ba4ab1b-e592-47f0-b059-1d7c202bc3c5)

The editing will have to be (for now) by manually editing the csv file

**Comment by rlisch** [comment]
Thanks Offer, this sounds good for a beginning.Best regards,RichardOn Jun 18, 2023, at 1:53 AM, Offer Shmuely ***@***.***> wrote:﻿
I maybe can show the total time on this page:

The editing will have to be (for now) by manually editing the csv file

**Comment by Shonkonray** [comment]
> I am still looking for this total flight time counter. The provided persistent counter only makes sense if it could be adjusted (through GV) to the real/actual total flight time of the model which I keep in my case, semi-automatic, using Flight Logs in an Excel file. Use it for maintenance of my turbines, gas engines and helis, very useful actually.

Hello @rlisch,

it is possible to do what you want with the persistent timer. I did so on my own yesterday!

1) Please set your timer as pfeerick has described.
2) Now load your model.yml from your transmitter on your computer (keep a back up, just in case!)
3) Open the .yml with a texteditor (notepad or similar)
4) Search for your set timer in this file and locate the entry "value"
5) Convert your total flight time records from days, hours, minutes, however you have saved them, to seconds and type this number into the field of "value"
6) Close and save the file and put it back on your SD card

This should do the trick, now you only have to choose your prefered widget.

Please see the attached files for how this could look like.

![Total_flight_timer_yml](https://github.com/EdgeTX/edgetx/assets/35926350/9f4892a9-51ed-48e2-8e84-b3b2e25469eb)
![Total_flight_timer](https://github.com/EdgeTX/edgetx/assets/35926350/3711c230-d2ee-400f-9412-bfbd4c1b5a45)

**Reply by rlisch** [reply]
Shonkonray, you made my day!!!
Thank you so much, this is exactly what I was looking for, and so simple. Took me less than an hour to add the correct Total Flight Time to each of my 58 models from my Excel flight logs.
Not beeing able to do this on the field is no issue at all as once the TFT is up do date nothing else has to be adjusted. It was a missing link on my elaborated model setups, specially for my helis and gassers, for maintenance purpose.
Again, thank you and best regards from the Caribbean.

**Comment by offer-shmuely** [comment]
very good and simple idea
although you can not do that on the field

**Reply by pfeerick** [reply]
Hm... can a Lua script/widget not manipulate the value? Although to be honest, it's probably quicker to do it on the computer, as you don't have to keep switching from model to model and then running the lua tool.

**Reply by Shonkonray** [reply]
I think the quickest and best way would be to allow more hours to be set via the special function "Set timer".
I dont know why this is limited to 8:59:59 [h:mm:ss]. Remove the limiter and all would be good. 👍 🤔

**Comment by yifeim** [comment]
Seems that this discussion has caught some attentions so I wanted to share
my approach. I set a toggle switch for log at 0.5 seconds. This switch is
combined with some other functions - throttle kill for powered planes and
vario for sailplanes. This way I get all the logs as well as time
information. I haven’t looked at the logs yet but I know that all the
information is there.

Still, displaying total time would be cool and convenient. It’s all about
simplicity.

**Reply by pfeerick** [reply]
Same here... I have logging linked to my arm logic (and also telemetry present), and it just keeps logging away until the time I want (or need) to look at it.

**Comment by Shonkonray** [comment]
> Shonkonray, you made my day!!! Thank you so much, this is exactly what I was looking for, and so simple. Took me less than an hour to add the correct Total Flight Time to each of my 58 models from my Excel flight logs. Not beeing able to do this on the field is no issue at all as once the TFT is up do date nothing else has to be adjusted. It was a missing link on my elaborated model setups, specially for my helis and gassers, for maintenance purpose. Again, thank you and best regards from the Caribbean.

You are welcome, glad it did help out. 😄 
In case others want to have the same, but do want to use the total flight time for something else, lets say program maintenance alarm every x-hours directly on your TX, you could also use a GV to count total hours. Please look at the screenshot below.
And this solution would work on the field too!

![Total_flight_GV](https://github.com/EdgeTX/edgetx/assets/35926350/ba70b966-c62e-4100-a269-17b854dd04ff)

**Comment by offer-shmuely** [comment]
Maybe I can write a script to set the timer to a specific day/hour/min
Will that help?

**Comment by rlisch** [comment]
Was at the field every day this week and Total Flight Time works perfect. Also installed the suggested method by SchonKonRay for Maintenance Intervalls, also works perfect.Added this to all my Helis for Thrust Bearing Maintenance on the feathering shaft which I have set to 2:00:00 flight time.Now if a specific date could be scheduled as well would be another great addition for gas plane setups, for example. I have scheduled for all gassers, no matter how many hours in the air, 12 months intervalls for changing all Tygon fuel lines.Appretiate what you are doing!

**Comment by offer-shmuely** [comment]
some preview (from the "PRESET" development:
Note: still very ugly GUI, but setting a persistent time is working

![image](https://github.com/EdgeTX/edgetx/assets/7026911/d92518bd-5ed1-45bc-a378-4cf442e5d8d5)

![image](https://github.com/EdgeTX/edgetx/assets/7026911/309e666a-aee3-4880-8060-95afa11c24c7)

![image](https://github.com/EdgeTX/edgetx/assets/7026911/21b3b051-1836-4718-b11d-3eddc76e232f)

**Comment by michaweb01** [comment]
Sorry, I feel plain stupid, but how can I download the files from this folder to be able to copy it on my SD Card?

https://github.com/EdgeTX/edgetx-sdcard/tree/master/sdcard/c480x272/SCRIPTS/TOOLS/FlightsHistory


Thanks
